### ISSUE TITLE
Shared Gem for editing a jekyll site hosted on GitHub pages

### ISSUE THREAD

**Issue by andy-wojciechowski** [msoe-sg/msoe-sg-editor-ui#17]
We can share a lot of our code with the SSE post editor if we include it in a shared library. What we can do is port over some of the backend from: https://github.com/msoe-sg/msoe-sg-editor into a gem. This should not have any references to rails or anything specific to SSE in this gem to prevent security risks when deploying.

- [x] Create a new gem project
- [x] Port over service classes from the archived msoe-sg-editor repo.
- [x]  Remove all references to Rails.configuration
- [x] Setup MiniTest
- [x] Port over unit tests for those service classes
- [x] Add rubocop
- [x] Add project to TravisCI
- [x] Figure out how to deploy Gem

**Comment by andy-wojciechowski** [comment]
@GFLEMING133 I know I wrote this in the UI project but I'm using this repo's issue system to track issues for the editor in general

**Comment by andy-wojciechowski** [comment]
@GFLEMING133 I wrote up an additional task in this issue for you related to the gem to remove all references to Rails.configuration which is a reference to environment specific config files in rails. Instead what I want you to do is replace those references like with an extra parameter or whatever is necessary to remove the reference.

Also for the unit tests I've added an additional task to setup the MiniTest framework in the gem. We cannot use ActiveSupport since that's a rails specific framework but I think they are very similar frameworks so that shouldn't be too much work. You might need to find a different way to run the tests since rails sets up rake automatically to run tests for us.

Finally, porting over the unit tests is going to be the longest part of this and the most tedious. I wrote a lot of tests for these classes and it's going to take a while to fix them so they no longer rely on rails. If you want I'd be happy to take porting some of the tests off your hands due to that being tedious if you think that's a good idea?.
